server/models/user.model.ts

```diff
@@ -0,0 +1,35 @@
+import { Schema, model, Document } from "mongoose";
+
+export interface IuserModel {
+  name: string;
+  email: string;
+  password: string;
+  access_token: string;
+}
+
+export interface IuserDocument extends IuserModel, Document {}
+
+const userModelSchema = new Schema<IuserDocument>(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    email: {
+      type: String,
```

Comment:
add validation function which validates email, is it in correct format or not

Reply:
this is implemented in middleware payload is verifying for /sign-up and /login end point